src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -157,4 +161,23 @@ public void deassignBook(@PathVariable Long userId, @PathVariable Long bookId) {
         userRepository.save(user);
     }
 
+    /**
+     * Update password of a {@link User}.
+     *
+     * @param id the user's id
+     * @param user the user
+     * @throws EntityNotFoundException
+     * @throws BookNotOwnedException
+     * @return the user
+     */
+    @PutMapping("/{id}/password")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void updatePassword(@PathVariable Long id, @RequestBody User user) {
+        if (user.getId() != id)
+            throw new IdMismatchException(User.class);
+        User dbUser = userRepository.findById(id).orElse(null);
+        dbUser.setPassword(encodePassword(user.getPassword()));
```

Comment:
I think that `.orElse(null);` could make everything to fail in the line 179, because we will send a message to null (`NullPointerException`)

Reply:
:O